Solutions/Corvus.Monitoring.Instrumentation.Abstractions/Corvus/Monitoring/Instrumentation/AdditionalInstrumentationDetail.cs

```diff
@@ -0,0 +1,34 @@
+﻿// <copyright file="AdditionalInstrumentationDetail.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Corvus.Monitoring.Instrumentation
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Additional information supplied with instrumentation.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Each of the supported instrumentation types can have additional information attached -
+    /// key value pairs that are either strings or numbers.
+    /// </para>
+    /// </remarks>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Usage",
+        "CA2227:Collection properties should be read only",
+        Justification = "TODO: needs some thought - we want it to be possible for either property to be null, and we want object initializer syntax to work so I'm not sure I agree with this warning")]
```

Comment:
We usually do this with a read-only property with an initialiser. Doesn't that turn the initialiser code into calls.yo `Add()` under the covers?

Reply:
I had forgotten that worked. For some reason I was under the impression that read-only properties were a barrier to object initializer expressions. (And I still think they are for non-collection properties, but I'd forgotten they special-cased collections.)